be/src/util/buffered_stream.cpp

```diff
@@ -149,4 +149,15 @@ void SharedBufferedInputStream::release() {
     _map.clear();
 }
 
+void SharedBufferedInputStream::release_to_offset(int64_t offset) {
+    auto it = _map.begin();
+    while (it != _map.end()) {
+        if (it->first <= offset) {
+            it = _map.erase(it);
```

Comment:
If use lower_bound/upper bound, and then use 'erase( iterator first, iterator last ) ',  it may be more effective as it can effectively reduces the container size by the number of elements removed.

Reply:
Thanks. That's better.